pkg/controller/vsphere/reconciler.go

```diff
@@ -575,28 +687,27 @@ func (vm *virtualMachine) getRegionAndZone(c *rest.Client, regionLabel, zoneLabe
 	return result, nil
 }
 
-func (vm *virtualMachine) reconcilePowerState() (bool, error) {
+func (vm *virtualMachine) reconcilePowerState() (bool, string, error) {
 	powerState, err := vm.getPowerState()
 	if err != nil {
-		return false, err
+		return false, "", err
 	}
 	switch powerState {
 	case types.VirtualMachinePowerStatePoweredOff:
 		klog.Infof("%v: powering on", vm.Obj.Reference().Value)
-		_, err := vm.powerOnVM()
+		task, err := vm.powerOnVM()
 		if err != nil {
-			return false, errors.Wrapf(err, "failed to trigger power on op for vm %q", vm)
+			return false, "", errors.Wrapf(err, "failed to trigger power on op for vm %q", vm)
 		}
-		// TODO: store task in providerStatus/conditions?
+
 		klog.Infof("%v: requeue to wait for power on state", vm.Obj.Reference().Value)
-		return false, nil
+		return false, task, nil
 	case types.VirtualMachinePowerStatePoweredOn:
 		klog.Infof("%v: powered on", vm.Obj.Reference().Value)
-	default:
-		return false, errors.Errorf("unexpected power state %q for vm %q", powerState, vm)
+		return true, "", nil
 	}
 
-	return true, nil
+	return false, "", errors.Errorf("unexpected power state %q for vm %q", powerState, vm)
```

Comment:
I'd be tempted to keep this as the default case for the switch, the compiler should realise no further return statement is needed

Reply:
👍